bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/ZkLedgerUnderreplicationManager.java

```diff
@@ -884,4 +885,39 @@ public String getReplicationWorkerIdRereplicatingLedger(long ledgerId)
         }
         return replicationWorkerId;
     }
+
+    @Override
+    public void setCheckAllLedgersCTime(long checkAllLedgersCTime) throws UnavailableException {
+        LOG.debug("setCheckAllLedgersCTime");
+        try {
+            if (zkc.exists(checkAllLedgersCtimeZnode, false) != null) {
+                zkc.setData(checkAllLedgersCtimeZnode, Long.toString(checkAllLedgersCTime).getBytes(UTF_8), -1);
+            } else {
+                zkc.create(checkAllLedgersCtimeZnode, Long.toString(checkAllLedgersCTime).getBytes(UTF_8),
+                        Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
+            }
+        } catch (KeeperException ke) {
+            throw new ReplicationException.UnavailableException("Error contacting zookeeper", ke);
+        } catch (InterruptedException ie) {
+            Thread.currentThread().interrupt();
+            throw new ReplicationException.UnavailableException("Interrupted while contacting zookeeper", ie);
+        }
+    }
+
+    @Override
+    public long getCheckAllLedgersCTime() throws UnavailableException {
+        LOG.debug("getCheckAllLedgersCTime");
+        try {
+            byte[] data = zkc.getData(checkAllLedgersCtimeZnode, false, null);
+            return Long.parseLong(new String(data, UTF_8));
```

Comment:
`Long.parseLong` will throw NumberFormatException. Can you catch that and return -1?

Reply:
added protobuf